subsys/net/ip/connection.c

```diff
@@ -777,6 +782,12 @@ static bool is_invalid_packet(struct net_pkt *pkt,
 		}
 	}
 
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_PACKET)) {
```

Comment:
Can you please add a code comment what happens here? (I suspect the root problem is that the function is misnamed, it should be "is_invalid_incoming_packet()" instead, but we have what we have, then comments are really helpful/needed.)